### PR TITLE
feat: skip thunder.torch modules

### LINKED ISSUE #475: Refine recording of source locations
From the discussion on #40 

@IvanYashchuk : 

If I use thunder.torch.softmax why is the source location recorded as

```
thunder/torch/__init__.py:4411: 	    return _softmax(a, dim=dim, dtype=dtype)
```

and with torch.softmax it points to the function being traced:

```
<ipython-input-14-72fa81ca31be>:2: 	    return torch.softmax(a, 0)
```

@t-vi 
This is because the "symbol boundary" for softmax is _softmax apparently (the symbols are opaque functions to the JIT components, so they are the topmost frame). Note that softmax is special in how we define it in thunder.torch (to cater for the undocumented parameter).

I see three options (for a follow-up PR):
- refuse to record locations in thunder.torch (deferring to the previous (next lower) stack frame if we are),
- provide a mechanism to fudge this (mark things "internal", might also be worthwile for "impl" functions if we see them),
- do nothing.

### PR DESCRIPTION
**Pull request by k223kim**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes #475.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

100%! It was cool to see how the interpreter work! 😎

### CODE REVIEW

**Review by t-vi — APPROVED** [review]
Supergood, thank you @k223kim !
